tests/routes/artifact_test.py

```diff
@@ -97,3 +90,17 @@ def test_illegal_artifact_upload(self):
             .route_params(bucket_name="test", path="_test") \
             .expect(403, self.RESPONSE_INVALID_NAME) \
             .post(data={"file": (StringIO("file contents"), "test.txt")}, headers=self.auth)
+
+    def test_bucket_configuration_doesnt_exist(self):
+        self.route_tester \
+            .artifact() \
+            .route_params(bucket_name="lol-this-doesnt-exist", path="hello/there") \
+            .expect(404, self.RESPONSE_404) \
+            .get(headers=self.auth)
+
+    def test_bucket_doesnt_exist(self):
+        self.route_tester \
+            .artifact() \
+            .route_params(bucket_name="thisBucketDoesntExistLol", path="hello/there") \
+            .expect(500, self.response_500()) \
+            .get(headers=self.auth)
```

Comment:
I am confused by this. if the configuration doesn't exist it throws a 404 and if the bucket doesn't exits it throws a 500. Seems like that is backwards and also both are mapped to `ErrorCode.RESOURCE_NOT_FOUND`.